src/pages/settings.tsx

```diff
@@ -0,0 +1,104 @@
+import { useCallback, useEffect } from 'react';
+
+import { openToast } from '@/components/lib/Toast';
+import { ComponentWrapperPage } from '@/components/near-org/ComponentWrapperPage';
+import { useBosComponents } from '@/hooks/useBosComponents';
+import { useDefaultLayout } from '@/hooks/useLayout';
+import { useAuthStore } from '@/stores/auth';
+import { useIdosStore } from '@/stores/idosStore';
+import type { NextPageWithLayout } from '@/utils/types';
+
+const SettingsPage: NextPageWithLayout = () => {
+  const components = useBosComponents();
+  const near = useAuthStore((store) => store.vmNear);
+  const idOS = useIdosStore((state) => state.idOS);
+  const idosUser = useIdosStore((state) => state.currentUser);
+  const idosCredentials = useIdosStore((state) => state.credentials);
+  const setIdosStore = useIdosStore((state) => state.set);
+
+  const connectIdOS = useCallback(async () => {
+    if (!near || !idOS) {
+      return;
+    }
+    const wallet = await (await near.selector).wallet();
+    await new Promise<void>(async (resolve, reject) => {
```

Comment:
I don't think we need this wrapping promise: `await new Promise<void>(async (resolve, reject) => {...})`. The wrapping `useCallback(async () => {...})` already turns this function body into a promise.

Reply:
After chatting with @shelegdmitriy, we agreed we can refactor this at a later time.